lib/fog/libvirt/models/compute/templates/server.xml.erb

```diff
@@ -93,6 +93,11 @@
     <channel type='unix'>
       <target type='virtio' name='org.qemu.guest_agent.0'/>
     </channel>
+<% end -%>
+<% if virtio_rng -%>
+    <rng model='virtio'>
+      <backend model='random'/>
```

Comment:
What about actually offering to provide the backend instead of relying on hypervisor default? As the documentation says "For QEMU, the default is /dev/random. However, the recommended source of entropy is /dev/urandom (as it doesn't have the limitations of /dev/random). "

Reply:
That's certainly possible. However, I'm unsure which default we should then use in `libfog`. Reading through https://bugzilla.redhat.com/show_bug.cgi?id=1074464 , it seems support for `/dev/urandom` has only been added in recent years. 
Both `/dev/random` and `/dev/urandom` can be problematic, though:
- `/dev/random` blocks if emptied by rogue VMs (but at least does not break quality), which in turn may break `/dev/urandom` in VMs. 
- `/dev/urandom` does not block if there is no entropy, potentially leading to bad randomness. 

The ideal solution would be to use `/dev/urandom` when enough entropy is available, and block otherwise. This is actually discussed in:
https://bugzilla.redhat.com/show_bug.cgi?id=1689946
and will be part of libvirt 6.1.0 via the `builtin` backend, which uses the `getrandom` syscall which has exactly this behaviour (c.f. https://github.com/libvirt/libvirt/commit/51489b84194b67be0d56820485de5aca555ebe97 ). But we can't rely on this yet, I think. 

So it is probably wise to expose both the `backend model` and then also the `backend path`, I agree. The question is: What should we default to, the most compatible, current default (`/dev/random`) or the recommended, but less compatible default (`/dev/urandom`)? 

Sorry, I really thought this PR was easier, but it turns out it's not...

Reply:
We should be ideally doing both as you said. I would introduce a simple array or hash to pass both values at once (no two settings). Default could be your current version, I don't see an issue with it, but at the same time it would mean that we use the default only when the user provided settings is wrong/unsupported, so maybe there don't have to be a default at all and don't have to limit what's provided either. Thoughts?

Reply:
I thought about it a bit and discussed with colleagues today. I think this is the best way to go, especially thinking about future changes to the defaults. I skimmed through `qemu` a bit and it seems they have moved to `/dev/urandom` as default [recently](https://github.com/qemu/qemu/commit/a2230bd778d8445231d2120b87be3ed262bb4f8a) and to `rng-builtin` (which uses `getrandom`) even more [recently](https://github.com/qemu/qemu/commit/0198c2621a1e57d4ceaca1326897f007963c12df). Libvirtd should automatically follow the changed defaults, so not having a default in libfog and not limiting anything seems to be the most reasonable. 
I'll give implementation and testing a go later :smile: .

Reply:
@strzibny What I pushed just now is completely untested (so there could be anything from syntax to logic errors) - I only pushed it to ask if this matches what you have been thinking of before I go ahead and test it in the next days.